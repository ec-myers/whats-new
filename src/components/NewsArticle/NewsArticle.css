.newsArticle {
  border-radius: 4px;
  border: 2px solid black;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  box-shadow: 0 0 10px #999999;
}

.articleImageDiv {
  height: 265px;
  width: 100%;
  overflow: hidden;
}

.articleImage {
  width: 100%;
  transform: translateY(-30px);
}

.articleBody {
  margin: 10px;
}

.articleFooter {
  background-color: papayawhip;
  height: 30px;
  padding: 5px;
  max-width: 100%;
}