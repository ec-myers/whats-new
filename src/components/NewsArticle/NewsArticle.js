import React from 'react';
import './NewsArticle.css';

const NewsArticle = article => {
  return (
    <article className="newsArticle">
      <div className="articleImageDiv">
        <img className="articleImage" src={article.img} alt="article" />
      </div>
      <h3 className="articleTitle">{article.headline}</h3>
      <p className="articleBody">{article.description}</p>
      <footer className="articleFooter">
        <a href={article.url} target="_blank">Read the Full Story</a>
      </footer>
    </article>
  )
}
export default NewsArticle;