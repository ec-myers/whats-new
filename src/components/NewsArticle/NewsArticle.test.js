import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  let article = {
    id: 1,
    headline: 'India\'s Lost Moon Lander Is Somewhere in This Photo',
    img: 'https://cdn.mos.cms.futurecdn.net/FsUeVGmY89BWeb36U8cUj5-650-80.jpg',
    description: 'The image was captured by NASA\'s Lunar Reconnaissance Orbiter (LRO) on Sept. 17 as it flew over the targeted landing site of India\'s Chandrayaan-2 mission.',
    url: 'https://www.livescience.com/india-moon-landing-site-nasa-photo.html',
  }
  it('should match the snapshot', () => {
    const wrapper = shallow(<NewsArticle
      key={article.id}
      headline={article.headline}
      img={article.img}
      url={article.url}
      description={article.description}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});