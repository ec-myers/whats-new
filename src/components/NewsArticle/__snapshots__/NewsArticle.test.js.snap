// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `
<article
  className="newsArticle"
>
  <div
    className="articleImageDiv"
  >
    <img
      alt="article"
      className="articleImage"
      src="https://cdn.mos.cms.futurecdn.net/FsUeVGmY89BWeb36U8cUj5-650-80.jpg"
    />
  </div>
  <h3
    className="articleTitle"
  >
    India's Lost Moon Lander Is Somewhere in This Photo
  </h3>
  <p
    className="articleBody"
  >
    The image was captured by NASA's Lunar Reconnaissance Orbiter (LRO) on Sept. 17 as it flew over the targeted landing site of India's Chandrayaan-2 mission.
  </p>
  <footer
    className="articleFooter"
  >
    <a
      href="https://www.livescience.com/india-moon-landing-site-nasa-photo.html"
      target="_blank"
    >
      Read the Full Story
    </a>
  </footer>
</article>
`;
